packages/babel-parser/src/tokenizer/index.js

```diff
@@ -1024,8 +1026,23 @@ export default class Tokenizer extends LocationParser {
 
     if (this.hasPlugin("bigInt")) {
       if (next === charCodes.lowercaseN) {
-        // disallow floats and legacy octal syntax, new style octal ("0o") is handled in this.readRadixNumber
-        if (isFloat || octal) this.raise(start, "Invalid BigIntLiteral");
+        // disallow numeric separators in non octal decimals
+        if (this.hasPlugin("numericSeparator") && isNonOctalDecimalInt) {
+          const underscorePos = this.input
```

Comment:
Could you move this whole `if` out of the `if (this.hasPlugin("bigInt"))`? It also applies to normal numbers:
```js
0_1;
```

Reply:
Sure! I thought it was only for bigInts, can i ask if that's because bigInt is no more going to be experimental?

Reply:
Done!